decompiler/frontend/binaryninja/handlers/unary.py

```diff
@@ -91,22 +94,13 @@ def _lift_zx_operation(self, instruction: MediumLevelILInstruction, **kwargs) ->
             )
         return self.lift_cast(instruction, **kwargs)
 
-    def _lift_load_struct(self, instruction: mediumlevelil.MediumLevelILLoadStruct, **kwargs) -> UnaryOperation:
-        """Lift a MLIL_LOAD_STRUCT_SSA instruction."""
-        return UnaryOperation(
-            OperationType.dereference,
-            [
-                BinaryOperation(
-                    OperationType.plus,
-                    [
-                        UnaryOperation(OperationType.cast, [self._lifter.lift(instruction.src)], vartype=Pointer(Integer.char())),
-                        Constant(instruction.offset),
-                    ],
-                    vartype=self._lifter.lift(instruction.src.expr_type),
-                ),
-            ],
-            vartype=Pointer(self._lifter.lift(instruction.src.expr_type)),
-        )
+    def _lift_load_struct(self, instruction: mediumlevelil.MediumLevelILLoadStruct, **kwargs) -> MemberAccess:
+        """Lift a MLIL_LOAD_STRUCT_SSA (struct member access e.g. var#n->x) instruction."""
+        struct_variable = self._lifter.lift(instruction.src)
+        struct_ptr: Pointer = self._lifter.lift(instruction.src.expr_type)
+        struct_type: Struct = struct_ptr.type
+        struct_member_name = struct_type.get_member_by_offset(instruction.offset).name
+        return MemberAccess(vartype=struct_ptr, operands=[struct_variable], offset=instruction.offset, member_name=struct_member_name)
```

Comment:
What do you mean by redundant calculation? Do you mean `instruction.offset` instead of  `member.offset` or  `struct_member_name = struct_type.get_member_by_offset(instruction.offset).name`?

I just did not want to pack everything into constructor line, so I opted for more local variables.

I can suggest the following alternative:
```python
struct_member = struct_ptr.type.get_member_by_offset(instruction.offset)
return MemberAccess(vartype=struct_ptr, operands=[struct_variable], offset=struct_member.offset, member_name=struct_member.name)
```
What do you think?

Reply:
perfect

Reply:
Done.